.github/workflows/upload-docs.yml

```diff
@@ -0,0 +1,36 @@
+name: Upload documentation
+on:
+  tags:
+  workflow_dispatch:
+  push:
+    branches:
+      - main
+
+jobs:
+  build:
+    if: github.repository_owner == 'Qiskit'
+    name: Build
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-python@v5
+        name: Install Python
+        with:
+          python-version: "3.11"
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          python -m pip install -r ./requirements-dev.txt
+
+      - name: Build documentation
+        run: python -m sphinx -W --keep-going docs/ docs/_build
+      - name: Store built documentation artifact
+        uses: actions/upload-artifact@v4
+        with:
+          name: qst-docs
+          path: |
+            ./docs/_build/*
+            !**/.doctrees
+            !**/.buildinfo
+          if-no-files-found: error
```

Comment:
Not a blocker: Since this code is the same as in the `ci.yml` workflow, could we reuse it to avoid repeated code?